test/m1_facilitator/repositories/GatewayRepository/util.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+
+
+import assert from '../../../test_utils/assert';
+import Gateway from '../../../../src/m1_facilitator/models/Gateway';
+
+const Util = {
+
+  assertGatewayAttributes(
+    inputGateway: Gateway,
+    expectedGateway: Gateway,
+  ): void {
+    assert.strictEqual(
+      inputGateway.gatewayGA,
+      expectedGateway.gatewayGA,
+      'Mismatch in gateway\'s global address.',
+    );
+
+    assert.strictEqual(
+      inputGateway.remoteGA,
+      expectedGateway.remoteGA,
+      'Mismatch in remote\'s global address',
+    );
+
+    assert.strictEqual(
+      inputGateway.gatewayType,
+      expectedGateway.gatewayType,
+      'Mismatch in type of gateway.',
+    );
+
+    assert.strictEqual(
+      inputGateway.destinationGA,
+      expectedGateway.destinationGA,
+      'Mismatch in destination global address.',
+    );
+
+    assert.strictEqual(
+      inputGateway.remoteGatewayLastProvenBlockNumber,
+      inputGateway.remoteGatewayLastProvenBlockNumber,
+      'Expected remote gateway\'s last proven block number is '
+      + `${inputGateway.remoteGatewayLastProvenBlockNumber} but got`
+      + `${expectedGateway.remoteGatewayLastProvenBlockNumber}.`,
+    );
+
+    assert.strictEqual(
+      inputGateway.anchorGA,
+      expectedGateway.anchorGA,
+      'Mismatch in anchor\'s global address.',
+    );
+
+    if (inputGateway.createdAt && expectedGateway.createdAt) {
```

Comment:
Why is this if condition needed?

Reply:
It was needed as `createdAt` object is `optional` in the model.
However, i see your point that it would fail when both are `undefined`.
So, i have checked it in the assertion and it is validated there.
https://github.com/mosaicdao/facilitator/pull/315/files#diff-896b96307547f0405df2c292cea4bf26R74